doc(partner_sdk): Update Dev guide with user tasks description (#61)
fivetran/fivetran_sdk: development-guide.md

- Partner code should handle any source/destination errors, retry any transient errors internally without deferring them to Fivetran.
- Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors instead of throwing exceptions and abruptly closing the connection.

### User Tasks

- Partners can throw alerts on the Fivetran dashboard to notify customers about potential issues with their connector.

> - These issues could include bad source data or connection problems with the source itself. The alerts can also provide guidance to customers on how to resolve the problem.

Reviewer: ```suggestion
- These issues could include bad source data or connection problems with the source itself. Where applicable, the alerts should also provide guidance to customers on how to resolve the problem.
```

- Currently, we allow only throwing [errors](https://fivetran.com/docs/using-fivetran/fivetran-dashboard/alerts#errors).

> - Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors and pass the message in the following json format to the error description:

Reviewer: ```suggestion
- Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors and pass the message in the following JSON format to the error description:
```

```
{
    "message": "Your task message goes here"
    "alert_type": "TASK"
}
```
- Usage example:
```
responseObserver.onError(Status.UNAVAILABLE.withDescription(jsonMessage).asException());
```

> - We convert the error description to a SEVERE log if it is not in the correct JSON format and fail the sync.

Reviewer: ```suggestion
- We convert the error description to a SEVERE log if it is not in the correct JSON format and the error causes the sync to fail.
```
?

Author: RIght, okay

- Partner code should handle any source/destination errors, retry any transient errors internally without deferring them to Fivetran.
- Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors instead of throwing exceptions and abruptly closing the connection.

### User Tasks

> - Partners can throw alerts on dashboard to notify customers when there is a problem in their Fivetran account, such as a broken connector or an incomplete sync, and tells you how to resolve that problem.

Reviewer 2: `Partners can throw alerts on the Fivetran dashboard to notify customers when there is a problem in their connector.`
And for the second line: the connector breaks because partners are throwing tasks and not vice versa.

Author: I corrected the first statement. According to me the examples are fine

Reviewer 2: We are the ones who break the connector if it is not working as expected by throwing the task. And that would cause incomplete sync.
e.g. Inappropriate value for a column (bad source), in this situation we break the connector intentionally and not that the connector broke so we are asking partners to throw a task. 

Updated doc can be: 
`Partners can throw alerts on the Fivetran dashboard to notify customers about potential issues with their connector. These issues could include bad source data or connection problems with the source itself. The alerts can also provide guidance to customers on how to resolve the problem.`

- Currently we allow only throwing [errors](https://fivetran.com/docs/using-fivetran/fivetran-dashboard/alerts#errors).
- Partner code should use [gRPC built-in error mechanism](https://grpc.io/docs/guides/error/#language-support) to relay errors and pass the message in the following json format to the error description:
```
{
    "message": "Your task message goes here"
    "alert_type": "TASK"
}
```

- Usage example:
```
responseObserver.onError(Status.UNAVAILABLE.withDescription(jsonMessage).asException());
```

> - The error description would be converted to a SEVERE log if it is not in the correct JSON format.

Reviewer 2: We are not just writing SEVERE log, we are even failing the connector right?

Author: Mentioned the same.
